presentation/backend/src/presentation-backend/Utils.ts

```diff
@@ -53,3 +54,150 @@ export interface BackendDiagnosticsAttribute {
   /** @alpha */
   diagnostics?: BackendDiagnosticsOptions;
 }
+
+/** @public */
+export type DiagnosticsCallback = (spans: ReadableSpan[]) => void;
+
+/**
+ * Mirrors the ReadableSpan interface from [opentelemetry-js](https://github.com/open-telemetry/opentelemetry-js).
+ * ReadableSpan[] can be passed to OpenTelemetry exporters.
+ * @public
+ */
+export interface ReadableSpan {
+  name: string;
+  kind: SpanKind;
+  spanContext: () => { traceId: string, spanId: string, traceFlags: number };
+  parentSpanId?: string;
+  startTime: HrTime;
+  endTime: HrTime;
+  status: { code: SpanStatusCode };
+  attributes: Attributes;
+  links: [];
+  events: TimedEvent[];
+  duration: HrTime;
+  ended: boolean;
+  resource: Resource;
+  instrumentationLibrary: { name: string };
+}
+
+/** @public */
+export type HrTime = [number, number];
+
+/** @public */
+export enum SpanStatusCode {
+  UNSET = 0,
+  OK = 1,
+  ERROR = 2
+}
+
+/** @public */
+export interface Attributes  {
+  [attributeKey: string]: string | string[];
+}
+
+/** @public */
+export interface TimedEvent {
+  time: HrTime;
+  name: string;
+  attributes: { [attributeKey: string]: string };
+}
+
+/** @public */
+export class Resource {
+  public attributes: Attributes;
+
+  constructor(attributes: Attributes) {
+    this.attributes = attributes;
+  }
+
+  public merge(other: Resource | null): Resource {
+    return new Resource({ ...this.attributes, ...other?.attributes });
+  }
+}
+
+/** @internal */
+export function convertToReadableSpans(diagnostics: Diagnostics): ReadableSpan[] {
```

Comment:
I believe it's not correct to duplicate OpenTelemetry types in our code. Instead, the `Diagnostics` to `ReadableSpan[]` conversion should be made by something that has appropriate OpenTelemetry dependencies. It seems the conversion code isn't very trivial, so maybe we should introduce a new `@itwin/presentation-opentelemetry` package? It could have the necessary type dependencies and would only have to be imported by consumers that are actually going to use OpenTelemetry (e.g. GPB). @calebmshafer, any objects to introducing a new package?